### PR TITLE
Fix Icspresso status queries

### PR DESCRIPTION
**Pull request by rmccue**
The ElasticPress integration filters any HTTP requests made to the Elasticsearch server. This includes Icspresso queries, which are made differently.

Icspresso sends some HTTP requests with an empty body, specifically an empty array. The signing code in the Elasticsearch Integration plugin doesn't account for array bodies, and this causes Request to eventually throw an exception. This causes a fatal, which is causing whitescreens when new sites are created and Icspresso tries to create an index for them.

This PR makes that code more robust, and uses the same mechanism of building the body as WP_Http (and Requests) uses internally. I've tested this in production (😱) and it allowed me to create the index for a new site which was fatalling without the patch.

tl;dr this should fix search breaking when new sites are added

cc @tcrsavage @joehoyle 

This may need backporting to `master`?

### CODE REVIEW

**Comment by joehoyle** [comment]
This will need backporting yeah
